Source/Container/HCList.h

```diff
@@ -30,6 +30,9 @@ typedef struct HCListIterator {
 } HCListIterator;
 extern const HCListIterator HCListIteratorInvalid;
 
+typedef void (*HCListForEachFunction)(HCRef value);
```

Comment:
Should ForEach offer the iteration index? For that matter, should we have iteration index for HCSet as well?